subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -78,11 +79,23 @@ public FileCollection getClasspath() {
         return super.getClasspath();
     }
 
+    /**
+     * The classpath for compiler plugin to run, for example, AST transformation.
+     *
+     * @since 5.6
+     */
     @Classpath
+    @Incubating
     public FileCollection getCompilerPluginClasspath() {
```

Comment:
Nothing criticial but I think I'd prefer _plural_ in both the name of the method and comments:

`getCompilerPluginsClasspath()`

_for example, AST transformations_.

Reply:
Personally I don't like the plural... what do people most likely type in the script? `compilerPluginClasspath` or `compilerPluginsClasspath`? I think `compilerPluginClasspath` is more natural to memorize. 

(Remember `allprojects` and `allProjects`?)

Reply:
The plural for me represents the reality. There's not one compiler plugin, there are many.

Reply:
I understand. For me, the word `classpath` implies that there're many _entries_. Similarly, we use `annotationProcessor`, not `annotationProcessors`, even that many annotation processors are possible, right?

Reply:
`annotationProcessor` was poorly chosen. We discussed this in the past and the reason it's named like that is only because Android had it. It should have been something like `annotationProcessing`, and have an `annotationProcessingClasspath`.

Reply:
@melix I am still wondering about the difference between compiler plugins and AST transformations. I thought AST transformations are passed via the compile classpath to the Groovy compiler, though it looks they are not:

https://github.com/gradle/gradle/blob/9d928bfdf9135154f7d484a6ac79e43b449e3fd3/subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/ApiGroovyCompiler.java#L110-L123

Do you know what classpath compiler plugins use? Do they also use the `transformClassLoader` or something else? What classes are available to compiler configuration scripts?

Reply:
Just spotted a very similar thing: https://github.com/gradle/gradle/blob/master/subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/precompiled/PrecompiledScriptPlugins.kt#L48

`kotlinCompilerPluginClasspath` is this the same/similar thing in Kotlin? Shall we unify these two terms?

Reply:
Sounds like a good idea. `groovyCompilerPluginClasspath` sounds like a good name for the configuration which will be introduced in #9516. The property should then stay being `compilerPluginClasspath`. I don't have a very strong opinion if it should be `compilerPluginClasspath` or `compilerPluginsClasspath`.

Regarding singular or plural the Gradle code base is not very consistent: `SourceTask.source`, `AbstractCompile.compileOptions`, `AbstractNativeCompileTask.getIncludes`.
We don't seem to have any `get*Classpath` method with a plural in the middle, though.